### ISSUE TITLE
Zed creates ~/.cache/zed and ignores XDG_CACHE_HOME on MacOS

### ISSUE THREAD

**Issue by ftpd** [zed-industries/zed#14277]
### Check for existing issues

- [X] Completed

### Describe the bug / provide steps to reproduce it

Zed creates `~/.cache/zed` after each run. It should rather use `XDG_CACHE_HOME`, which in my case is set to /tmp/<username> (which doesn't really matter).

### Environment

Zed: v0.143.6 (Zed)
OS: macOS 15.0.0
Memory: 18 GiB
Architecture: aarch64

### If applicable, attach your Zed.log file to this issue.

<details><summary>Zed.log</summary><pre>

</pre></details>

**Comment by notpeter** [comment]
Yep. Looks like we are not checking `XDG_CACHE_HOME` on linux and are only checking `FLATPAK_XDG_CACHE_HOME`:

https://github.com/zed-industries/zed/blob/489077befcc1ac4e1aa4505fa3f27f1a6528da48/crates/paths/src/paths.rs#L58-L79

Shouldn't be too hard to fix. Happy to accept PRs.

**Comment by someone13574** [comment]
> Yep. Looks like we are not checking `XDG_CACHE_HOME` on linux and are only checking `FLATPAK_XDG_CACHE_HOME`:
> 
> https://github.com/zed-industries/zed/blob/489077befcc1ac4e1aa4505fa3f27f1a6528da48/crates/paths/src/paths.rs#L58-L79
> 
> Shouldn't be too hard to fix. Happy to accept PRs.

That snippet calls `dir::cache_dir()` which [reads the env variable](https://docs.rs/dirs/5.0.1/dirs/fn.cache_dir.html). `FLATPAK_XDG_CACHE_HOME` is only used to override this when running in flatpak. I don't see why this wouldn't work other than if the environment variable is being unset somehow.

**Comment by notpeter** [comment]
Sorry I had misread your specs. I've retagged it as MacOS. Nice catch.

I see your point, if we're going to act XDG-like and use `~/.cache/` by default we should consider allowing XDG_CACHE_HOME to override. But I think we are unlikely to respect XDG_CACHE_HOME on MacOS.

It might be appropriate to switch to `dirs::cache_dir()` as I believe already use `~/Library/Caches/dev.zed.Zed` elsewhere.  Alternatively we could just use `$TMPDIR`.

Anyways, I think I understand your request.

**Comment by ftpd** [comment]
> MacOS doesn't use XDG environment variables, so it is expected that it wouldn't be followed.

It's not a task for operating system anyway. Programs running on it do or don't use these variables. I'm setting `XDG_*_HOME` on both Linux and macOS and I expect software I run to follow it  ;-)

`~/Library/Caches` as a 'hardcoded' value fits me too. Everything but `~/.cache` ;-)

**Comment by kilingzhang** [comment]
Is there any update on this? I also hope to follow this specification or set environment variables directly within the ZED software. Since ZED currently lacks multi-end synchronization for configuration and other data, we can achieve configuration synchronization by synchronizing data within this directory.
